src/Compilers/CSharp/Test/Semantic/Semantics/BindingAsyncTests.cs

```diff
@@ -4009,17 +4020,20 @@ public class C : I
                 Diagnostic(ErrorCode.WRN_UnobservedAwaitableExpression, "i.M()").WithLocation(11, 1),
                 // (12,1): warning CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.
                 // i.MAsync(); // 5
-                Diagnostic(ErrorCode.WRN_UnobservedAwaitableExpression, "i.MAsync()").WithLocation(12, 1)
+                Diagnostic(ErrorCode.WRN_UnobservedAwaitableExpression, "i.MAsync()").WithLocation(12, 1),
+                // (20,5): warning CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.
+                //     c.MAsync(); // 6
```

Comment:
For some reason I thought we didn't make this suggestion because it's bad to await in a lock.

Reply:
For `async` method we consider it's always okay to await, regardless of context :-/ This isn't great, but it's an existing issue/question, orthogonal to what I'm fixing.